backend/api/models.py

```diff
@@ -245,6 +245,18 @@ def has_perm(self, perm, obj=None):
         # return True for now
         return True
 
+    def can_add_user(self, user, store, new_user_role):
+        try:
+            assoc = Association.objects.get(user=user, store=store)
+        except Association.DoesNotExist:
+            return False
+        if assoc.role == Role.EMPLOYEE:
+            return False
+        elif assoc.role == Role.MANAGER:
+            return new_user_role in [Role.MANAGER, Role.EMPLOYEE]
+        elif assoc.role == Role.VENDOR:
+            return True
```

Comment:
Role Based Permissions for Adding New User

Reply:
Doesn't Matt's Endpoint already cover this?

Reply:
Matt's recently created an endpoint that allows one to change someone's role. This one creates the employee/manager/vendor and associates them to a store.